django_datadog_logger/middleware/request_id.py

```diff
@@ -11,22 +11,17 @@ def generate_request_id():
 
 def get_or_create_request_id(request):
     request_id = request.META.get("HTTP_X_REQUEST_ID")
-    if request_id and re.match("^[a-zA-Z0-9+/=-]{20,200}$", request_id):
-        return request_id
-    else:
-        return generate_request_id()
+    return request_id or generate_request_id()
 
 
-class RequestIdMiddleware:
-    def __init__(self, get_response=None):
-        self.get_response = get_response
+class RequestIdMiddleware(MiddlewareMixin):
 
-    def __call__(self, request):
+    def process_request(self, request):
         request.request_id = get_or_create_request_id(request)
         request.request_start_time = time.time()
         local.request = request
-        response = self.get_response(request)
+
+    def process_response(self, request, response):
         response["X-Request-ID"] = request.request_id
-        if hasattr(local, "request"):
-            del local.request
```

Comment:
cleanup of request is required, otherwise this stale data might be left pending for next requests